ivadomed/main.py

```diff
@@ -378,6 +384,75 @@ def run_command(context, n_gif=0, thr_increment=None, resume_training=False):
                 nib.save(pred, os.path.join(pred_path, filename))
 
 
+def create_dataset_and_ivadomed_version_log(context):
+
+    dataset_paths = context['loader_parameters']['bids_path']
+
+    ivadomed_version = imed_utils._version_string()
+    datasets_version = []
+
+    if isinstance(dataset_paths, str):
+        datasets_version = [imed_utils.__get_commit(path_to_git_folder=dataset_paths)]
+    elif type(dataset_paths)==type([]):  # Bad
+        for Dataset in dataset_paths:
+            datasets_version.append(imed_utils.__get_commit(path_to_git_folder=Dataset))
+
+    log_file = os.path.join(context['log_directory'], 'version_info.log')
+
+    try:
+        f = open(log_file, "w")
+    except OSError as err:
+        print("OS error: {0}".format(err))
+        raise Exception("Have you selected a log folder you have writing permissions?")
+
+    # IVADOMED
+    f.write('IVADOMED TOOLBOX\n----------------\n(' + ivadomed_version + ')')
+
+    # DATASETS
+    f.write('\n\n\nDATASET VERSION\n---------------\n')
+    if len(datasets_version) == 1:
+        f.write('A single BIDS dataset was used for training.\n' + dataset_paths)
+        if datasets_version[0] != '?!?':  # This is what was returned from _version_string when no git is present
+            f.write(' - Dataset Annex version: ' + datasets_version[0] + '\n\n')
+        else:
+            f.write(' - Dataset is not Annexed.\n')
+    else:
+        f.write('The following BIDS datasets were used for training.\n')
+
+        for iDataset in len(dataset_paths):
+            if len(datasets_version[0]) != 0:
+                f.write(str(iDataset+1) + '. ' + dataset_paths[iDataset] + ' - Dataset Annex version: ' + datasets_version[0] + '\n')
+            else:
+                f.write('\n')
+
+    # SYSTEM INFO
+    f.write('\n\nSYSTEM INFO\n-------------\n')
```

Comment:
I know we don't officially support Windows at the moment due to some bugs, should we add a check for `os_running = 'windows'` for when we add Windows support in the future? Or cross that bridge when we get there?

Reply:
yes we should!
Thanks for the suggestion

Reply:
No problem!!